sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java

```diff
@@ -350,9 +352,14 @@ SQLiteDatabase getDatabase() {
         try {
             return mSQLiteOpenHelper.getWritableDatabase();
         } catch (RuntimeException e) {
+            AppCenterLog.warn(LOG_TAG, "Failed to open database. Trying to delete database (may be corrupted).", e);
 
             /* First error, try to delete database (may be corrupted). */
-            mContext.deleteDatabase(mDatabase);
+            if (mContext.deleteDatabase(mDatabase)) {
```

Comment:
We should document this behavior to let customers know that we only ever delete data to recover from this fatal error scenario.